pages/add/form/validator.js

```diff
@@ -53,6 +58,39 @@ const Validator = {
         }
       }
     };
+  },
+  imageSizeValidator() {
+    return {
+      error: `File size is over 2MB.`,
+      validate: function(value) {
+        if (!value) {
+          return;
+        }
+
+        // there's no file size limit on the backend
+        // but it's still good that we enforce a size limit (2MB) on client side
+        let base64String = value.base64;
+        let sizeLimit = Math.pow(2, 20) * 2; // 2MB in bytes
```

Comment:
Yea I thought about it but couldn't decide. For many devs (like myself) `2097152` isn't a number that we can easily relate to 2MB. I used `Math.pow(2, 20) * 2` so that if we ever need to adjust the size limit in the future, we can simply change a number there and let it compute the final number.